drivers/sx127x/sx127x.c

```diff
@@ -0,0 +1,477 @@
+ *               2017 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_sx127x
+ * @{
+ * @file
+ * @brief       Basic functionality of sx127x driver
+ *
+ * @author      Eugene P. <ep@unwds.com>
+ * @author      José Ignacio Alamos <jose.alamos@inria.cl>
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ * @}
+ */
+#include <stdbool.h>
+#include <math.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include "xtimer.h"
+#include "thread.h"
+
+#include "periph/gpio.h"
+#include "periph/spi.h"
+
+#include "sx127x.h"
+#include "sx127x_internal.h"
+#include "sx127x_registers.h"
+#include "sx127x_netdev.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+/* Internal functions */
+static void _init_isrs(sx127x_t *dev);
+static void _init_timers(sx127x_t *dev);
+static int _init_peripherals(sx127x_t *dev);
+static void _on_tx_timeout(void *arg);
+static void _on_rx_timeout(void *arg);
+
+/* SX127X DIO interrupt handlers initialization */
+static void sx127x_on_dio0_isr(void *arg);
+static void sx127x_on_dio1_isr(void *arg);
+static void sx127x_on_dio2_isr(void *arg);
+static void sx127x_on_dio3_isr(void *arg);
+
+
+void sx127x_setup(sx127x_t *dev, const sx127x_params_t *params)
+{
+    netdev_t *netdev = (netdev_t*) dev;
+    netdev->driver = &sx127x_driver;
+    memcpy(&dev->params, params, sizeof(sx127x_params_t));
+}
+
+void sx127x_reset(sx127x_t *dev)
+{
+    /*
+     * This reset scheme complies with 7.2 chapter of the SX1272/1276 datasheet
+     * See http://www.semtech.com/images/datasheet/sx1276.pdf for SX1276
+     * See http://www.semtech.com/images/datasheet/sx1272.pdf for SX1272
+     *
+     * 1. Set NReset pin to LOW for at least 100 us
+     * 2. Set NReset in Hi-Z state
+     * 3. Wait at least 5 milliseconds
+     */
+    gpio_init(dev->params.reset_pin, GPIO_OUT);
+
+    /* Set reset pin to 0 */
+    gpio_clear(dev->params.reset_pin);
+
+    /* Wait 1 ms */
+    xtimer_usleep(1000);
+
+    /* Put reset pin in High-Z */
+    gpio_init(dev->params.reset_pin, GPIO_IN);
+
+    /* Wait 10 ms */
+    xtimer_usleep(1000 * 10);
+}
+
+int sx127x_init(sx127x_t *dev)
+{
+    sx127x_reset(dev);
+
+    /* Do internal initialization routines */
+    if (!_init_peripherals(dev)) {
+        return -SX127X_ERR_SPI;
+    }
+
+    _init_isrs(dev);
+    _init_timers(dev);
+
+    xtimer_usleep(1000); /* wait 1 millisecond */
+
+    /* Check presence of SX127X */
+    if (!sx127x_test(dev)) {
+        DEBUG("[Error] init : sx127x test failed\n");
+        return -SX127X_ERR_TEST_FAILED;
+    }
+
+    sx127x_rx_chain_calibration(dev);
+
+    /* Set RegOpMode value to the datasheet's default.
+       Actual default after POR is 0x09 */
+    sx127x_reg_write(dev, SX127X_REG_OPMODE, 0x00);
+
+    return SX127X_INIT_OK;
+}
+
+void sx127x_init_radio_settings(sx127x_t *dev)
```

Comment:
A new question again !
Why some variables are initialized with netdev and the rest with the driver functions ?